dataland-email-service/src/test/kotlin/org/dataland/datalandemailservice/services/EmailUnsubscriberTest.kt

```diff
@@ -0,0 +1,68 @@
+package org.dataland.datalandemailservice.services
+
+import org.dataland.datalandemailservice.email.EmailSender
+import org.dataland.datalandemailservice.entities.EmailSubscriptionEntity
+import org.dataland.datalandemailservice.repositories.EmailSubscriptionRepository
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when`
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doNothing
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.springframework.http.ResponseEntity
+import java.util.UUID
+
+class EmailUnsubscriberTest {
+    private lateinit var emailSender: EmailSender
+    private lateinit var internalEmailBuilder: InternalEmailBuilder
+    private lateinit var emailUnsubscriber: EmailUnsubscriber
+    private lateinit var emailSubscriptionRepository: EmailSubscriptionRepository
+
+    private val validUuid = UUID.randomUUID()
+    private val emailAddress = "test@example.com"
+    private val invalidUuid = UUID.randomUUID()
+
+    private val validEmailSubscriptionEntity =
+        EmailSubscriptionEntity(validUuid, emailAddress, true)
+
+    @BeforeEach
+    fun setup() {
+        emailSender = mock(EmailSender::class.java)
+        internalEmailBuilder = mock(InternalEmailBuilder::class.java)
+        emailUnsubscriber = mock(EmailUnsubscriber::class.java)
+        emailSubscriptionRepository = mock(EmailSubscriptionRepository::class.java)
+
+        emailUnsubscriber =
+            EmailUnsubscriber(
+                emailSubscriptionRepository,
+                emailSender,
+                internalEmailBuilder,
+            )
+
+        `when`(emailSubscriptionRepository.findByUuid(validUuid)).thenReturn(validEmailSubscriptionEntity)
+        `when`(emailSubscriptionRepository.findByUuid(invalidUuid)).thenReturn(null)
+        `when`(internalEmailBuilder.buildInternalEmail(any())).thenReturn(mock())
+        doNothing().whenever(emailSender).filterReceiversAndSentEmail(any())
+    }
+
+    @Test
+    fun `should unsubscribe email and send notification when email exists`() {
```

Comment:
declaration doesn't explain what is testet; please rephrase

Reply:
I am not sure what you expected but I changed it.